source/features/deep-blame.tsx

```diff
@@ -0,0 +1,106 @@
+import './deep-blame.css';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import React from 'dom-chef';
+import features from '../libs/features';
+import {getRepoGQL, getReference} from '../libs/utils';
+import versionIcon from 'octicon/versions.svg';
+
+const filterMergeCommits = async (commits: string[]): Promise<AnyObject> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			${commits.map((commit: string) => `
+				${api.escapeKey(commit)}: object(expression: "${commit}") {
+					... on Commit {
+						associatedPullRequests(last: 1) {
+							nodes {
+								mergeCommit {
+									oid
+								}
+								commits(last: 1) {
+									nodes {
+										commit {
+											oid
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			`).join('\n')}
+		}
+	`);
+
+	const lastCommits = new Map();
+	for (const [key, commit] of Object.entries<AnyObject>(repository)) {
+		const {nodes} = commit.associatedPullRequests;
+		const mergeCommit = nodes[0].mergeCommit.oid;
+		const lastCommit = nodes[0].commits.nodes[0].commit.oid;
+		if (mergeCommit === key.slice(1)) {
+			lastCommits.set(mergeCommit, lastCommit);
+		}
+	}
+
+	return lastCommits;
+};
+
+function getCommitHash(commit: HTMLElement): string {
+	return (commit.nextElementSibling! as HTMLAnchorElement).href.split('/').slice(-1)[0];
+}
+
+async function init(): Promise<void | false> {
+	const pullRequests = select.all('[data-hovercard-type="pull_request"]');
+	if (pullRequests.length === 0) {
+		return;
+	}
+
+	const commits = await filterMergeCommits([...new Set(pullRequests.map(getCommitHash))]);
```

Comment:
I thought we agreed to make this action only appear AFTER alt clicking, no API calls should be made before. It's in my first post,

> 2. When the button is clicked, it has to query the API to get the PR's latest commit (user will be shown a loading icon)

Reply:
Understand.

However most times the version icon is not there even that you can go back?! Where should the use click?

Reply:
> Unless you mean don't run anything unless the user clicks a button and then load everything I wrote.

That’s exactly what I meant/wrote. 

My reasoning is that blame pages could have dozens of commits so the query could be very large. Also because this feature isn’t meant to be used that frequently, so even if the query is light, it would be useless.

Reply:
You asked for confirmation here, I confirmed, and then you changed the whole thing 🤷‍♂️ 

https://github.com/sindresorhus/refined-github/issues/2876#issuecomment-596212371

Reply:
First we are only showing for pull request commits

Second 
>However most times the version icon is not there even that you can go back?! Where should the use click?